CONTRIBUTING.md

```diff
@@ -0,0 +1,103 @@
+* **Include screenshots and animated GIFs** which help you demonstrate the steps. You can use [this tool](https://www.cockos.com/licecap/) to record GIFs on macOS and Windows, and [this tool](https://github.com/colinkeenan/silentcast) or [this tool](https://github.com/GNOME/byzanz) on Linux.
+* **Explain why this enhancement would be useful** to most users.
+
+### Your First Code Contribution
+
+Unsure where to begin contributing to uqcsbot? You can start by looking through these `beginner` and `help-wanted` issues:
+
+* [Entry level issues](https://github.com/UQComputingSociety/uqcsbot/issues?q=is%3Aissue+is%3Aopen+label%3A%22entry+level%22) - issues which should only require a few lines of code, and a test or two.
+* [Help wanted issues](https://github.com/UQComputingSociety/uqcsbot/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22) - issues which should be a bit more involved than `beginner` issues.
+
+#### Local development
+
+See the [README](README.md) for getting started on local development.
+
+### <a name="submit-pr"></a> Pull Requests
+
+1. Fork the UQComputingSociety/uqcsbot repo.
+1. Make your changes in a new git branch:
+
+     ```shell
+     git checkout -b my-fix-branch
+     ```
+
+1. Create your patch, **including test cases where appropriate**.
+1. Run the full test suite, as described in the [README](README.md),
+  and ensure that all tests pass.
+1. Commit your changes using a descriptive commit message.
+
+     ```shell
+     git add .
+     git commit -m "descriptive message"
+     ```
+
+1. Push your branch to GitHub:
+
+    ```shell
+    git push origin my-fix-branch
+    ```
+
+1. In GitHub, send a pull request to `uqcsbot:python-rewrite`.
+    * If we suggest changes then:
+    * Make the required updates.
+    * Re-run the test suite to ensure tests are still passing.
+
+1. If the PR is a small addition, one approval should suffice before merging. Else for more complicated PRs, multiple approvals may be necessary.
+
+That's it! Thank you for your contribution!
+
+## Styleguides
+
+### Git Commit Messages
+
+* Use the present tense ("Add feature" not "Added feature")
+* Reference issues and pull requests liberally after the first line
+* Consider starting the commit message with an applicable emoji:
+    * :art: `:art:` when improving the format/structure of the code
+    * :racehorse: `:racehorse:` when improving performance
+    * :memo: `:memo:` when writing docs
+    * :bug: `:bug:` when fixing a bug
+    * :fire: `:fire:` when removing code or files
+    * :white_check_mark: `:white_check_mark:` when adding tests
+    * :lock: `:lock:` when dealing with security
+    * :arrow_up: `:arrow_up:` when upgrading dependencies
+    * :arrow_down: `:arrow_down:` when downgrading dependencies
+    * :shirt: `:shirt:` when removing linter warnings
+
+### Python Styleguide
+
+All Python code must adhere to the [PEP-8 Styleguide](https://www.python.org/dev/peps/pep-0008/) as closely as possible.
```

Comment:
Maybe merge this section into the Styleguides section above?

Reply:
Done :^)

Reply:
(well, in-lined it into the PR submission section)